### PR TITLE
Data sync functionality.

### LINKED ISSUE #4: Sql relations services not configured

### LINKED ISSUE #12: DataSync test

### PR DESCRIPTION
**Pull request by gyKa**
closes #12 and closes #4

### CODE REVIEW

File: Tests/Functional/Overall/DataSyncTest.php [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+        // Set initial start position, if not sure, 0 always returns results.
+        $this->setLastSync(0, BinlogParser::START_TYPE_POSITION);
+
+        // Set service so, that it would use last sync position.
+        $this
+            ->getServiceContainer()
+            ->get('ongr_connections.sync.diff_provider.binlog_diff_provider')
+            ->setStartType(BinlogParser::START_TYPE_POSITION);
+
+        // Then start data import with pipeline $target.
```
Comment (trylika) [inline Tests/Functional/Overall/DataSyncTest.php:55]: data provide, not data import

File: Tests/Functional/Overall/DataSyncTest.php [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+use ONGR\ConnectionsBundle\Sync\DiffProvider\Binlog\BinlogParser;
+use ONGR\ElasticsearchBundle\DSL\Search;
+use ONGR\ElasticsearchBundle\ORM\Repository;
+
+class DataSyncTest extends TestBase{
+
+    /**
+     * Tests data importing from client data to sync storage.
+     */
+    public function testDataImportFromClientDataToSyncStorage()
```
Comment (trylika) [inline Tests/Functional/Overall/DataSyncTest.php]: functionality changed for this test, we should rename this test and it's annotation

File: Tests/Functional/Overall/DataSyncTest.php (hunk 1 above)
Comment (gyKa) [inline Tests/Functional/Overall/DataSyncTest.php:55]: "data provide" is name of process, I think "data import" discribes this process better.

File: Tests/Functional/Fixtures/DataImportTest/updateProducts.sql [hunk 3]
```diff
@@ -1,2 +1,11 @@
 UPDATE `oxarticles` SET
-  `OXSHORTDESC` = 'The same description for all!';
+  `OXTITLE` = 'The same title for all!',
+  `OXSHORTDESC` = 'The same desc for all!';
```
Comment (mindaugasbarysas) [inline Tests/Functional/Fixtures/DataImportTest/updateProducts.sql]: Why not description instead of desc?

**Pull request merged**